File: src/test/java/com/outsmart/DataGeneratorTest.java
package com.outsmart;

import org.junit.Test;

import static junit.framework.Assert.*;

/**
 * @author Vadim Bobrov
 */
public class DataGeneratorTest {

    private DataGenerator dataGenerator = new DataGenerator();

    @Test
    public void testGetRandomCustomer() throws Exception {
        assertTrue(dataGenerator.getRandomCustomer().startsWith("customer"));
    }

    @Test
    public void testGetCustomer() throws Exception {
        for(int i = 0; i < 20; i++)
            assertEquals(dataGenerator.getCustomer(i), "customer" + i);
    }

    @Test
    public void testGetRandomLocation() throws Exception {
        assertTrue(dataGenerator.getRandomLocation().startsWith("location"));
    }

    @Test
    public void testGetLocation() throws Exception {
        for(int i = 0; i < 2; i++)
            assertEquals(dataGenerator.getLocation(i), "location" + i);
    }

    @Test
    public void testGetRandomWireId() throws Exception {
        assertTrue(dataGenerator.getRandomWireId().startsWith("wire"));
    }

    @Test
    public void testGetWireId() throws Exception {
        for(int i = 0; i < 300; i++)
            assertEquals(dataGenerator.getWireId(i), "wire" + i);
    }

    @Test
    public void testGetRandomMeasurement() throws Exception {
        assertTrue(dataGenerator.getRandomMeasurement() > 0);
    }
}

File: src/test/scala/com/outsmart/DataGeneratorTest.scala
package com.outsmart

import org.scalatest.FunSuite

/**
 * @author Vadim Bobrov
*/
class DataGeneratorTest extends FunSuite {

  test("random customer") {
    val dataGen = new DataGen
    assert(dataGen.getRandomCustomer.startsWith("customer"))
  }

  test("select customer") {
    val dataGen = new DataGen
    for(i <- 0 until 20)
      assert(dataGen.getCustomer(i).equals("customer" + i))
  }

  test("random location") {
    val dataGen = new DataGen
    assert(dataGen.getRandomLocation.startsWith("location"))
  }

  test("select location") {
    val dataGen = new DataGen
    for(i <- 0 until 2)
      assert(dataGen.getLocation(i).equals("location" + i))
  }

  test("random wireid") {
    val dataGen = new DataGen
    assert(dataGen.getRandomWireId.startsWith("wireid"))
  }

  test("select wireid") {
    val dataGen = new DataGen
    for(i <- 0 until 300)
      assert(dataGen.getWireId(i).equals("wireid" + i))
  }

  test("random measurement") {
    val dataGen = new DataGen
    assert(dataGen.getRandomMeasurement > 0)
  }

}
